### ISSUE TITLE
Disable keyboard navigation looping

### ISSUE THREAD

**Issue by gonfunko** [RaspberryPiFoundation/blockly#9496]
Currently, navigating through blocks with the keyboard will eventually reach the "end" of the workspace and loop around to the first block stack, forever. Instead, when the end is reached, navigation should stop and an audible cue should be emitted. The audio portion may require some experimentation and validation to identify an approach that is compatible/audible in conjunction with a screenreader.

**Comment by BenHenning** [comment]
Note that this changes the behavior for regular keyboard navigation users, too. I wonder if this necessitates adding jumping shortcuts for stack navigation or moving to the top/bottom stack of the workspace to allow for the same looping behavior to exist without it being default enabled.

**Comment by gonfunko** [comment]
See #9498 for the shortcuts :) My inclination here is to allow toggling looping on the `LineCursor` instance to keep backwards compatibility; then the keyboard experiment can just disable it, but we can keep the current behavior for other users, and if we decide we want it back it's just a method call away to revert.

**Comment by kmcnaught** [comment]
Just to flag that we didn't used to have looping on keyboard navigation, and it was a big confusion in user testing, particularly in ambiguous/confusing layouts like this (everyone pressed right arrow, nothing happened, and eventually most admitted defeat, possibly after trying down arrow to no avail too). Adding looping means that _something_ always happens, so even if the ordering is a bit weird you get some feedback and find a way back to where you want. 

<img width="1422" height="589" alt="empty forever block to the left of a taller on start stack, centre aligned so that the forever block is lower and technically after the tall stack" src="https://github.com/user-attachments/assets/dc51501e-7bd1-4e2e-9cd5-71b504c8cdc7" />

**Comment by gonfunko** [comment]
Noted! With this layout and the current navigation right/down does navigate you to the next stack, even with looping turned off. The looping toggle is also effectively a one-liner, so this should be easy to revert if testing suggests that's needed or even if we want to vary the behavior contextually.

**Comment by kmcnaught** [comment]
@gonfunko I think you misunderstood the example. I can still replicate the issue in [screenreader demo](https://raspberrypifoundation.github.io/blockly-keyboard-experimentation/screenreader/) (assuming that's the correct deployment?) with the following code. With the cursor in the position shown in the screenshot, `right arrow` takes you to the icon field first, and then `down arrow`/`right arrow` both result in an error tone and no position change.

<img width="1794" height="1318" alt="draw heart block on left, taller repeat while block on the right but taller so the top is higher" src="https://github.com/user-attachments/assets/a3e3dd3e-b044-42ac-bab8-1bcd84262d19" />

The issue is that the ordering of the blocks is based on a left->right and top->bottom weighted ordering based on the x,y position of a block (top left I believe). When blocks/stacks on the right are taller than those on the left, this gives very unintuitive ordering. The point is that whatever way[1] we project stack positions down to a 1D ordering, there will be cases that are ambiguous or unintuitive, and looping was introduced in [#369](https://github.com/RaspberryPiFoundation/blockly-keyboard-experimentation/issues/369) to resolve this for sighted users. 

I'm keen that we don't lose what we gained from keyboard testing. If you are now mainly focusing on screen reader users, you are less likely to come up against this confusion. As to how you meet both groups' needs, I'm not sure - I'm not involved in any of the screen reader work. 

[1] Note that the same ordering also impacts execution order (in some platforms) so even if we had an ordering that was less often confusing, we can't easily change the existing behaviour, and wouldn't want keyboard/screen reader ordering to differ from execution order.

**Comment by gonfunko** [comment]
Ah, yes I had set up things slightly wrong when I tried it, sorry about that! I also very much agree that we don't want to have things regress. I talked this over with the team and our general plan at the moment is to continue as-is in the screenreader branch while we experiment and try things there, but at the point when we're ready to merge changes back into mainline we'll need to address this (and other features, like the level-change beeps). That may be through making them configurable or otherwise refactoring/fixing them to be satisfactory for all types of users. Thank you for flagging this too!
